src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWTH_RATE = 1.5;
+    private int size;
+    private Object[] array;
+
+    public ArrayList() {
+        array = new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+            return;
+        }
+        newCapacity();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < size) {
+            if ((size + 1) > array.length) {
+                newCapacity();
+            }
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+            size++;
+        } else {
```

Comment:
this else block is redundant
if (check) {
throw new ....
}
will be enough

Reply:
Fixed!